packages/@ember/-internals/glimmer/lib/helpers/or.ts

```diff
@@ -0,0 +1,30 @@
+import { Arguments, CapturedArguments, VM } from '@glimmer/runtime';
+import { InternalHelperReference } from '../utils/references';
+
+/**
+@module ember
+*/
+
+/**
+  Evaluates the given arguments with || in short-circuit
+
+  Example:
+
+  ```handlebars
+  {{or isAdmin isAuthor}}
+
+  {{! be validData either `isAdmin` or `isAuthor` are truthy}}
+  ```
+
+  @public
+  @method or
+  @for Ember.Templates.helpers
+  @since 2.7.0
+*/
+function or({ positional: { references } }: CapturedArguments) {
+  return references.reduce((acc: any, ref) => acc || ref.value(), false);
```

Comment:
This could return early too if references is an array.
Using `references.some` while checking for any reference value to be true would return early as or only cares for any true argument.